### PR TITLE
Add close method and fixed some issues

### PR DESCRIPTION
**Pull request by jimut**
List of changes are:

1. Added close method.
2. Removed return statement in constructor.
3. Async function should not explicitly return a promise.
4. Removed unnecessary catch statement.
5. Removed async callback inside forEach.
6. Changed forEach async operation to sync for loop.

Please review and merge the changes.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -127,57 +123,59 @@ class RedisContentful {
 
       // Deleting all the deleted entries from redis
       if (response.deletedEntries && response.deletedEntries.length) {
-        response.deletedEntries.forEach(async entry => {
+        for (let i = 0; i < response.deletedEntries.length; i += 1) {
```
Comment (shreyas-a) [inline src/index.js]: if `forEach` was not the issue can you please replace it back with `for` loop?

Comment (shreyas-a) [reply src/index.js]: everything else LGTM 👍🏼

Comment (jimut) [reply src/index.js]: It is not causing an issue in this particular use case but can potentially skip deleting deleted entries in redis in the long run.

Comment (shreyas-a) [reply src/index.js]: I think `for...of` would be appropriate here! Check [this](https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop)!

Comment (jimut) [reply src/index.js]: Okay.

**Comment by jimut** [comment]
Check now.

**Review by shreyas-a — APPROVED** [review]

**Pull request merged**